keras_cv/layers/preprocessing/resizing.py

```diff
@@ -0,0 +1,287 @@
+    BaseImageAugmentationLayer,
+)
+
+H_AXIS = -3
+W_AXIS = -2
+
+
+class Resizing(BaseImageAugmentationLayer):
+    """A preprocessing layer which resizes images.
+
+    This layer resizes an image input to a target height and width. The input
+    should be a 4D (batched) or 3D (unbatched) tensor in `"channels_last"`
+    format.  Input pixel values can be of any range (e.g. `[0., 1.)` or `[0,
+    255]`) and of interger or floating point dtype. By default, the layer will
+    output floats.
+
+    This layer can be called on tf.RaggedTensor batches of input images of
+    distinct sizes, and will resize the outputs to dense tensors of uniform
+    size.
+
+    For an overview and full list of preprocessing layers, see the preprocessing
+    [guide](https://www.tensorflow.org/guide/keras/preprocessing_layers).
+
+    Args:
+      height: Integer, the height of the output shape.
+      width: Integer, the width of the output shape.
+      interpolation: String, the interpolation method. Defaults to `"bilinear"`.
+        Supports `"bilinear"`, `"nearest"`, `"bicubic"`, `"area"`, `"lanczos3"`,
+        `"lanczos5"`, `"gaussian"`, `"mitchellcubic"`.
+      crop_to_aspect_ratio: If True, resize the images without aspect
+        ratio distortion. When the original aspect ratio differs from the target
+        aspect ratio, the output image will be cropped so as to return the
+        largest possible window in the image (of size `(height, width)`) that
+        matches the target aspect ratio. By default
+        (`crop_to_aspect_ratio=False`), aspect ratio may not be preserved.
+    """
+
+    def __init__(
+        self,
+        height,
+        width,
+        interpolation="bilinear",
+        crop_to_aspect_ratio=False,
+        pad_to_aspect_ratio=False,
+        bounding_box_format=None,
+        **kwargs,
+    ):
+        self.height = height
+        self.width = width
+        self.interpolation = interpolation
+        self.crop_to_aspect_ratio = crop_to_aspect_ratio
+        self.pad_to_aspect_ratio = pad_to_aspect_ratio
+        self._interpolation_method = keras_cv.utils.get_interpolation(interpolation)
+        self.bounding_box_format = bounding_box_format
+
+        if pad_to_aspect_ratio and crop_to_aspect_ratio:
+            raise ValueError(
+                "`Resizing()` expects either `crop_to_aspect_ratio` or "
+                "`pad_to_aspect_ratio`, but not both."
+            )
+        if crop_to_aspect_ratio and bounding_box_format:
+            # TODO(lukewood): support `bounding_box.smart_resize()`
+            raise ValueError(
+                "Resizing() does not support `crop_to_aspect_ratio=True` "
+                "and `bounding_box_format` at the same time.  In order to resize with "
+                "bounding boxes, please pass `crop_to_aspect_ratio=False`."
+            )
+        super().__init__(**kwargs)
+
+    def _augment(self, inputs):
```

Comment:
Is it really necessary to override _augment and _batch_augment for this layer?

Can't we just do the normal _augment_image, _augment_bounding_boxes, etc?

Reply:
Then instead of having these big methods e.g. resize_with_pad, individual methods like augment_image could delegate to image-specific resizing functionality based on crop/pad settings

Reply:
For now lets keep this given that it matches `keras.layers.Resizing` and that this layer is used in many inference time pipelines.  We want to make sure we preserve the performance of these inference pipelines, so by avoiding unexpected vmaps and map_fns we can ensure that we don't regress on performance.